py3status/modules/diskdata.py

```diff
@@ -141,51 +248,44 @@ def diskdata(self):
             "full_text": self.py3.safe_format(self.format, self.values),
         }
 
-    def _get_free_space(self, disk):
-        if disk and not disk.startswith("/dev/"):
-            disk = "/dev/" + disk
+    def _get_free_space(self):
 
         total = 0
         used = 0
         free = 0
         devs = []
 
-        df = self.py3.command_output("df")
-        for line in df.splitlines():
-            if (disk and line.startswith(disk)) or (
-                disk is None and line.startswith("/dev/")
-            ):
-                data = line.split()
-                if data[0] in devs:
+        disk_names = [d.name for d in self._disks]
+        _df = self.py3.command_output("df")
+        # loop on df output minus header line
+        for line in _df.splitlines()[1:]:
+            df = Df(*line.split())
+            if self._resolve_disk_name(df.device) in disk_names:
+                if df.device in devs:
```

Comment:
This fails on first device. `if sda in []`.

Reply:
can you please explain? I see that if i put something wrong that does not begin with a /:

```
FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/block/sdd/queue/hw_sector_size'
  File "/home/oggei/git/py3status-src/py3status/module.py", line 142, in prepare_module
    self.module_class.post_config_hook()
  File "diskdata.py", line 193, in post_config_hook
    self._add_monitored_disk("/dev/" + item)
  File "diskdata.py", line 112, in _add_monitored_disk
    with open("{}/queue/hw_sector_size".format(block_device)) as ss:

```

do you mean this? Or what?

Reply:
I've fixed this meanwhile

Reply:
you know, it's a bit frustrating to not have a proper testing framework different to edit last lines of a module and then remembering not to commit those changes. I think a way to at least specify config via CLI would be useful. Maybe I'll contribute something like that

Reply:
Free feel to contribute! :smile: 

I usually use this way (less hassle).
```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index 5eea41d..d730403 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -92,6 +92,9 @@ class Py3status:
     types = ["disks"]
     unit = "B/s"
 
+    # playtime
+    self.disks = ["sdd"]
+
     class Meta:
         deprecated = {
             "remove": [{"param": "sector_size", "msg": "obsolete parameter"}],
@@ -303,7 +306,6 @@ if __name__ == "__main__":
     """
     Run module in test mode.
     """
-    config = {"disks": ["sdd"]}
     from py3status.module_test import module_test
 
-    module_test(Py3status, config=config)
+    module_test(Py3status)
```

Reply:
that's a clever suggestion, thanks

Reply:
>you know, it's a bit frustrating to not have a proper testing framework different to edit last lines of a module and then remembering not to commit those changes. I think a way to at least specify config via CLI would be useful. Maybe I'll contribute something like that

https://github.com/lasers/py3status/tree/module_test-cli-arguments

* `python3 diskdata.py --term --disk sdb`  # works okay cuz of your deprecation
* `python3 diskdata.py --disks "['sdb', 'sda']" --term`  # works okay too

The issue could be that we won't remember what we have or what to use... and it could be hassle to specify more than one... or that some configs can be hard to copy/specify like `--format`.

* `python3 loadavg.py --format "1min [\?color=cyan {1min}]\|[\?color=1avg {1avg}%]" --term`

Our shell history would be filled with random test strings too. I think it is easier to override this inside the module instead, but it could be easier to specify this via CLI in some cases, but yeah, have a play with this and see how you feel about this feature. :feelsgood: